src/Simplex/Chat/Types.hs

```diff
@@ -230,70 +234,267 @@ defaultChatSettings = ChatSettings {enableNtfs = True}
+    CFReceipts -> receipts
+    CFVoice -> voice
+
+-- collection of optional chat preferences for the user and the contact
+data Preferences = Preferences
+  { fullDelete :: Maybe Preference,
+    receipts :: Maybe Preference,
+    voice :: Maybe Preference
   }
   deriving (Eq, Show, Generic, FromJSON)
 
-defaultChatPrefs :: ChatPreferences
-defaultChatPrefs = ChatPreferences {voice = Just Preference {enable = PSOff}}
+data GroupPreferences = GroupPreferences
+  { fullDelete :: Maybe GroupPreference,
+    receipts :: Maybe GroupPreference,
+    voice :: Maybe GroupPreference
+  }
+  deriving (Eq, Show, Generic, FromJSON)
+
+instance ToJSON GroupPreferences where
+  toJSON = J.genericToJSON J.defaultOptions {J.omitNothingFields = True}
+  toEncoding = J.genericToEncoding J.defaultOptions {J.omitNothingFields = True}
+
+instance ToField GroupPreferences where
+  toField = toField . encodeJson
+
+instance FromField GroupPreferences where
+  fromField = fromTextField_ decodeJson
+
+-- full collection of chat preferences defined in the app - it is use to ensure we include all preferences and to simplify processing
+-- if some of the preferences are not defined in Preferences, defaults from defaultChatPrefs are used here.
+data FullPreferences = FullPreferences
+  { fullDelete :: Preference,
+    receipts :: Preference,
+    voice :: Preference
+  }
+  deriving (Eq)
+
+-- merged preferences of user for a given contact - they differentiate between specific preferences for the contact and global user preferences
+data ContactUserPreferences = ContactUserPreferences
+  { fullDelete :: ContactUserPreference,
+    receipts :: ContactUserPreference,
+    voice :: ContactUserPreference
+  }
+  deriving (Show, Generic)
+
+data ContactUserPreference = ContactUserPreference
+  { enabled :: PrefEnabled,
+    userPreference :: ContactUserPref,
+    contactPreference :: Preference
+  }
+  deriving (Show, Generic)
+
+data ContactUserPref = CUPContact {preference :: Preference} | CUPUser {preference :: Preference}
+  deriving (Show, Generic)
+
+instance ToJSON ContactUserPreferences where toEncoding = J.genericToEncoding J.defaultOptions
 
-emptyChatPrefs :: ChatPreferences
-emptyChatPrefs = ChatPreferences {voice = Nothing}
+instance ToJSON ContactUserPreference where toEncoding = J.genericToEncoding J.defaultOptions
 
-instance ToJSON ChatPreferences where
+instance ToJSON ContactUserPref where
+  toJSON = J.genericToJSON . sumTypeJSON $ dropPrefix "CUP"
+  toEncoding = J.genericToEncoding . sumTypeJSON $ dropPrefix "CUP"
+
+toChatPrefs :: FullPreferences -> Preferences
+toChatPrefs FullPreferences {fullDelete, receipts, voice} =
+  Preferences
+    { fullDelete = Just fullDelete,
+      receipts = Just receipts,
+      voice = Just voice
+    }
+
+defaultChatPrefs :: FullPreferences
+defaultChatPrefs =
+  FullPreferences
+    { fullDelete = Preference {allow = PSNo},
+      receipts = Preference {allow = PSNo},
+      voice = Preference {allow = PSNo}
```

Comment:
I think voice should be enabled by default - seems like most users want it for one reason or another, and not all will have patience to find / set up preferences. "No" disables full feature here, where in other cases it's "sub feature", e.g. full delete doesn't disable deletion altogether, and receipts we don't even have and it's not really necessary for communication + both of these probably have higher impact on privacy/control of the recipient.
```suggestion
      voice = Preference {allow = PSYes}
```